src/goto-symex/goto_symex_state.cpp

```diff
@@ -439,10 +437,7 @@ bool goto_symex_statet::l2_thread_read_encoding(
 
     if(a_s_read.second.empty())
     {
-      auto level2_it =
-        level2.current_names.emplace(l1_identifier, std::make_pair(ssa_l1, 0))
-          .first;
-      symex_renaming_levelt::increase_counter(level2_it);
+      level2.increase_generation(l1_identifier, ssa_l1, path_storage);
```

Comment:
I think you should pass through `path_storage` as an argument, not make it a member of `goto_symex_statet`. This will increase the churn I'm afraid, but otherwise the coupling becomes too difficult to manage. While doing so, you actually shouldn't pass in all of `path_storage`, but just the bit that you really need.

Reply:
@tautschnig you'll then need to pass it to every call, and every recursive call, to `rename`, since that calls thread-read-encoding which in turn allocates a fresh L2 name. Are you sure that's better?

Reply:
That's what we are doing with field sensitivity... I actually do think that's better, but if you don't want to go as far then at least only a very limited part of `path_storaget` should be passed in. Otherwise we're creating a monolith. It's the sort of thing that @LAJW and @romainbrenguier have been arguing against in #4282.